lib/sync.js

```diff
@@ -72,6 +72,7 @@ function SyncApi(client, opts) {
     this._running = false;
     this._keepAliveTimer = null;
     this._connectionReturnedDefer = null;
+    this._notifEvents = []; // accumulator of sync events in the current sync response
```

Comment:
It would be easier to review this stuff if you didn't mix everything up together :/